cluster-agent/metrics/metrics.go

```diff
@@ -0,0 +1,75 @@
+package metrics
+
+import (
+	"fmt"
+	"time"
+
+	metric "sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/redhat-appstudio/managed-gitops/backend-shared/db"
+)
+
+var (
+	OperationStateCompleted = promauto.NewGauge(
+		prometheus.GaugeOpts{
+			Name:        "operationDB_completedState",
+			Help:        "Number of Operation DB rows that were set to completed state in the last hour",
+			ConstLabels: map[string]string{"OperationDBState": "completed"},
+		},
+	)
+
+	OperationStateFailed = promauto.NewGauge(
+		prometheus.GaugeOpts{
+			Name:        "operationDB_failedState",
+			Help:        "Number of Operations DB rows currently in non-complete state",
+			ConstLabels: map[string]string{"OperationDBState": "failed"},
+		},
+	)
+)
+
+func SetOperationDBState(state db.OperationState) {
+
+	if state == db.OperationState_Completed {
+		OperationStateCompleted.Inc()
+	} else if state == db.OperationState_Failed {
+		OperationStateFailed.Inc()
+	}
+
+	ticker := time.NewTicker(60 * time.Minute)
+
+	// Creating channel using make
+	tickerChan := make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-tickerChan:
+				return
+			case <-ticker.C:
+				// Reset counter to zero every 1 hour
+				ClearMetrics()
+			}
+		}
+	}()
+
+	// Calling Stop() method to stop the ticker
```

Comment:
Yes It never clears the metrics, I will add the Sleep time

Reply:
I have added Sleep method here, but unit test is failing because of timeout error, Is there any way to fix this?

Reply:
Why do we want to stop the timer anyway?  The Jira says that the metrics should be cleared every hour.

Reply:
So we don't need to stop the ticker and it will run indefinitely every hour , It stops when we stops our code right?
  something like this which run indefinitely every hour to clear the metrics
    ticker := time.NewTicker(time.Second)

	```
go func() {
		for {
			<-ticker.C

			ClearMetrics()

		}
	}()
        // using select here to run indefinitely
	select {}
```

Reply:
We shouldn't need the `select{}`, I've suggested a new contents for this function in my review comments.

Reply:
I have few questions regarding the pseudocode you have mentioned in the story, PTAL @jgwest